base/linalg/lq.jl

```diff
@@ -88,22 +88,25 @@ end
 
 size(A::LQPackedQ) = size(A.factors)
 
-full(A::LQ) = A[:L]*A[:Q]
-#=
-We construct the full eye here, even though it seems ineffecient, because
-every element in the output matrix is a function of all the elements of
-the input matrix. The eye is modified by the elementary reflectors held
-in A, so this is not just an indexing operation. Note that in general
-explicitly constructing Q, rather than using the ldiv or mult methods,
-may be a wasteful allocation.
-=#
-function full{T}(A::LQPackedQ{T}; thin::Bool=true)
+convert(::Type{Matrix}, A::LQ) = convert(Array, A[:L]*A[:Q])
+convert(::Type{Array}, A::LQ) = convert(Matrix, A)
+full(A::LQ) = convert(Array, A)
+
+function convert{T}(::Type{Matrix}, A::LQPackedQ{T}; thin::Bool = true)
```

Comment:
Cheers, that sounds wise. Thoughts:

`full` has keyword arguments only for `QRPackedQ`, `QRCompactWYQ`, and `LQPackedQ` types, each of which represents the orthogonal factor in the corresponding factorization. The corresponding factorizations are of the thin variety, and by default `full` on the orthogonal factor returns the thin orthogonal factor in `Matrix` form. That behavior seems natural and correct, i.e. the result is faithful to the factorization. The keyword argument allows the user to request an expanded orthogonal factor, which strikes me as less natural and correct though potentially occasionally useful. Perhaps having `convert` just return the thin form, and if necessary documenting how the user can recover an expanded orthogonal factor somewhere (a oneliner), would be a reasonable approach? For the purposes of smooth deprecation, we could have the relevant `full` methods called without a keyword argument dispatch to `convert`,  and define a version with keyword argument in `deprecated.jl` that issues a depwarn and note on how to retrieve an expanded orthogonal factor? @andreasnoack? Thanks again, and best!